support-models/src/main/scala/com/gu/support/workers/redemption/RedemptionData.scala

```diff
@@ -0,0 +1,35 @@
+package com.gu.support.workers.redemption
+
+import cats.syntax.functor._
+import com.gu.support.encoding.Codec
+import com.gu.support.zuora.api.{ReaderType, Subscription}
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder}
+
+abstract class RedemptionData(val redemptionCode: RedemptionCode) {
+  def redeem(subscription: Subscription) = subscription.copy(redemptionCode = Some(redemptionCode))
+}
+
+case class Corporate(override val redemptionCode: RedemptionCode, corporateAccountId: CorporateAccountId) extends RedemptionData(redemptionCode) {
+  override def redeem(subscription: Subscription) =
+    super.redeem(subscription)
+      .copy(
+        corporateAccountId = Some(corporateAccountId),
```

Comment:
this and redemptionCode are new (non/indexed?) fields in zuora Subscription object? or just internal model

Reply:
(I can see below that they are zuora custom fields)